crates/sparrow-physical/src/step.rs

```diff
@@ -0,0 +1,59 @@
+use arrow_schema::SchemaRef;
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[repr(transparent)]
+#[serde(transparent)]
+pub struct StepId(usize);
+
+/// A single step in the physical plan.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct Step {
```

Comment:
From a naming perspective:

"steps" describe a "table" or "batch" (set of columns) and "expressions" describe a column. Not sure if anyone has a better / clearer suggestion here.

Reply:
Personally, slight preference towards batch than table

Reply:
Agreed on that -- but the naming question here is `step` and `expression` (vs some alternative).

Reply:
`step` / `expression` looks good to me